testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java

```diff
@@ -65,46 +66,76 @@ public void testCreateDropSchema()
     }
 
     @Test
-    public void testDropSchemaDeletesDirectory()
+    public void testDropSchemaWithEmptyLocation()
     {
-        String schemaName = "schema_with_location";
+        String schemaName = schemaName("schema_with_empty_location");
         String schemaDir = warehouseDirectory + "/schema-with-location/";
 
-        ensureSchemaDoesNotExist(schemaName);
-
-        hdfsClient.createDirectory(schemaDir);
+        createSchema(schemaName, schemaDir);
+        assertFileExists(schemaDir, true, "schema directory exists after create schema");
+        assertQuerySucceeds("DROP SCHEMA " + schemaName);
+        assertFileExists(schemaDir, false, "schema directory exists after drop schema");
+    }
 
-        assertQuerySucceeds(format("CREATE SCHEMA %s WITH (location = '%s')", schemaName, schemaDir));
-        assertQuerySucceeds(format("DROP SCHEMA %s", schemaName));
+    @Test
+    public void testDropSchemaWithoutLocation()
```

Comment:
I'm not completely sure how this cast still works. If the location property is not set, `HiveMetadata` shouldn't be deleting anything, and the `HiveMetastore` implementations shouldn't be deleting anything here, either.

(Though I don't believe this test runs with every `HiveMetastore` we have, which would probably be good coverage to have.)